draft-ietf-quic-manageability.md

```diff
@@ -181,17 +181,201 @@ used to separate QUIC packets. The length header field is variable length and
+are generally coalesced (see {{coalesced}}) in order to reduce the number of UDP
+datagrams sent during the handshake.
+
+As shown here, the client can send 0-RTT data as soon as it has sent its Client
+Hello, and the server can send 1-RTT data as soon as it has sent its Server
+Hello.
+
+~~~~~
+Client                                    Server
+  |                                          |
+  +----QUIC Client Hello-------------------->|
+  +----(zero or more 0RTT)------------------>|
+  |                                          |
+  |<--------------------QUIC Server Hello----+
+  |<---------(1RTT encrypted data starts)----+
+  |                                          |
+  +----Initial Completion------------------->|
+  +----(optional 0RTT completion)----------->|
+  +----(1RTT encrypted data starts)--------->|
+  |                                          |
+  |<-----------------Handshake Completion----+
+  |                                          |
+~~~~~
+{: #fig-handshake
+   title="General communication pattern visible in the QUIC handshake"}
+
+A typical handshake starts with the client sending of a QUIC Client Hello
+datagram as shown in {{fig-client-hello}}, which elicits a QUIC Server Hello
+datagram as shown in {{fig-server-hello}} typically containing three packets:
+an Initial packet with the Server Hello, a Handshake packet with the rest of
+the server's side of the TLS handshake, and initial 1-RTT data, if present.
+
+The content of QUIC Initial packets are encrypted using Initial Secrets, which
+are derived from a per-version constant and the client's destination connection
+ID; they are therefore observable by any on-path device that knows the
+per-version constant; we therefore consider these as visible in our
+illustration. The content of QUIC Handshake packets are encrypted using keys
+established during the initial handshake exchange, and are therefore not
+visible.
+
+Initial, Handshake, and the Short Header packets transmitted after the handshake
+belong to cryptographic and transport contexts. The Initial Completion
+{{fig-init-complete}} and the Handshake Completion {{fig-hs-complete}} datagrams
+finish these first two contexts, by sending the final acknowledgment and
+finishing the transmission of CRYPTO frames.
+
+~~~~~
++----------------------------------------------------------+
+| UDP header (source and destination UDP ports)            |
++----------------------------------------------------------+
+| QUIC long header (type = Initial, Version, DCID, SCID) (Length)
++----------------------------------------------------------+  |
+| QUIC CRYPTO frame header                                 |  |
++----------------------------------------------------------+  |
+| TLS Client Hello (incl. TLS SNI)                         |  |
++----------------------------------------------------------+  |
+| QUIC PADDING frame                                       |  |
++----------------------------------------------------------+<-+
```

Comment:
This sets a dangerous pattern.  I am going to strongly recommend that people a. Put PADDING frames before CRYPTO, and b. Split CRYPTO frames.

Also there are multiple PADDING frames.